src/VisualStudio/Core/Def/Implementation/ProjectSystem/Legacy/AbstractLegacyProject.cs

```diff
@@ -163,7 +175,13 @@ public virtual void Disconnect()
             if (!string.IsNullOrEmpty(linkMetadata))
             {
                 var linkFolderPath = Path.GetDirectoryName(linkMetadata);
-                folders = linkFolderPath.Split(PathSeparatorCharacters).ToImmutableArray();
+                folders = linkFolderPath.Split(PathSeparatorCharacters, StringSplitOptions.RemoveEmptyEntries).ToImmutableArray();
+            }
+            else if (!string.IsNullOrEmpty(VisualStudioProject.FilePath))
```

Comment:
Probably move the comment for `_projectDirectory` here? It wasn't clear to me when I first looked at the code here why we check the file path from VisualStudioProject but save and use the folder path in `AbstractLegacyProject`

Reply:
Oh, I see. Would switching this to string.IsNullOrEmpty(_projectDirectory) clear it up since then this is consistent?